### PR TITLE
[PaymentBundle] Use a single encryption method

### PR DESCRIPTION
**Pull request by Wojdylak**
| Q               | A
|-----------------|-----
| Branch?         | 2.0
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | #17362
| License         | MIT

### CODE REVIEW

**Review by mpysiak — DISMISSED** [review]

File: src/Sylius/Behat/Page/Admin/PaymentMethod/UpdatePage.php [hunk 1]
```diff
@@ -89,6 +94,7 @@ protected function getDefinedElements(): array
             'secret_key' => '[data-test-secret-key]',
             'signature' => '[data-test-signature]',
             'username' => '[data-test-username]',
+            'use_payum' => '[data-test-use-payum]',
```
Comment (GSadee) [inline src/Sylius/Behat/Page/Admin/PaymentMethod/UpdatePage.php]:
```suggestion
            'use_payum' => '[data-test-use-payum]',
            'username' => '[data-test-username]',
```

File: src/Sylius/Bundle/PayumBundle/Tests/Checker/GatewayConfigEncryptionCheckerTest.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        $gatewayConfig = $this->createMock(GatewayConfigInterface::class);
+        $decoratedChecker = $this->createMock(BaseGatewayConfigEncryptionCheckerInterface::class);
+        $decoratedChecker->method('isEncryptionEnabled')->willReturn(false);
+
+        $checker = new GatewayConfigEncryptionChecker($decoratedChecker);
+        $this->assertFalse($checker->isEncryptionEnabled($gatewayConfig));
+    }
+
+    /** @test */
+    public function it_cannot_encrypt_when_gateway_config_use_payum(): void
```
Comment (GSadee) [inline src/Sylius/Bundle/PayumBundle/Tests/Checker/GatewayConfigEncryptionCheckerTest.php]:
```suggestion
    public function it_cannot_encrypt_when_gateway_config_uses_payum(): void
```

File: src/Sylius/Component/Payment/spec/Encryption/PaymentRequestEncrypterSpec.php [hunk 3]
```diff
@@ -130,8 +130,33 @@ function it_does_nothing_when_decrypting_payment_request_with_no_payload_and_emp
         $paymentRequest->getResponseData()->willReturn([]);
 
         $encrypter->decrypt(Argument::any())->shouldNotBeCalled();
+        $paymentRequest->setResponseData(Argument::any())->shouldNotBeCalled();
 
-        $paymentRequest->setResponseData([])->shouldBeCalled();
+        $this->decrypt($paymentRequest);
+    }
+
+    function it_does_not_decrypt_payment_request_is_not_string(
```
Comment (GSadee) [inline src/Sylius/Component/Payment/spec/Encryption/PaymentRequestEncrypterSpec.php]:
```suggestion
    function it_does_not_decrypt_if_payment_request_is_not_string(
```

File: config/packages/test/_sylius.yaml [hunk 4]
```diff
@@ -16,3 +16,8 @@ sylius_state_machine_abstraction:
         sylius_payment_request: '%test_sylius_state_machine_adapter%'
         sylius_product_review: '%test_sylius_state_machine_adapter%'
         sylius_shipment: '%test_sylius_state_machine_adapter%'
+
+sylius_payment:
+    encryption:
+        disabled_for_factories:
+            - online-disabled
```
Comment (GSadee) [inline config/packages/test/_sylius.yaml:23]: Is it needed? 🤔

**Review by GSadee — DISMISSED** [review]

File: config/packages/test/_sylius.yaml (hunk 4 above)
Comment (Wojdylak) [reply config/packages/test/_sylius.yaml:23]: Yes, for this test https://github.com/Sylius/Sylius/pull/17370/files#diff-a1a057c72a2184c796813b76a9f320c4d628bcd471b66d42ab47f2af89775ff0R56

**Review by GSadee — APPROVED** [review]

**Pull request merged**